### ISSUE TITLE
Multicluster controlPlaneSecurity--remote sidecars unable to connect to ADS

### ISSUE THREAD

**Issue by tiswanso** [istio/istio#7010]
**Describe the bug**
Remote cluster pod sidecars unable to connect to ADS in deployment of Istio multicluster with control-plane security enabled (sidecar-to-pilot = ADS over mTLS).

**Expected behavior**
Remote cluster pod sidecars able to connect to istio-pilot on main cluster via mTLS to access ADS. 

**Steps to reproduce the bug**
1.  Main cluster deployment:

    ```
        cat install/kubernetes/namespace.yaml > install/kubernetes/istio-auth.yaml
	$(HELM) template --set global.tag=${TAG} \
		--namespace=istio-system \
		--set global.hub=${HUB} \
		--values install/kubernetes/helm/istio/values.yaml \
		--set global.mtls.enabled=true \
		--set global.controlPlaneSecurityEnabled=true \
		--set security.selfSigned=false \
		install/kubernetes/helm/istio >> install/kubernetes/istio-auth.yaml
    ```
    1.   **NOTE** the `security.selfSigned=false`--the cacerts secret is created as follows:
          ```
           kubectl create secret generic cacerts -n istio-system --from-file=samples/certs/ca-cert.pem --from-file=samples/certs/ca-key.pem --from-file=samples/certs/root-cert.pem --from-file=samples/certs/cert-chain.pem
          ```
1.  Remote cluster setup:

     ```
      helm template install/kubernetes/helm/istio-remote --name istio-remote --namespace istio-system  --set global.remotePilotAddress=192.170.154.237 --set global.remotePolicyAddress=192.170.154.246 --set global.proxy.envoyStatsd.host=192.170.154.235 --set global.proxy.envoyStatsd.enabled=true --set global.ingressEndpoint=192.170.154.240 --set global.ingressGatewayEndpoint=192.170.154.222 --set global.remoteTelemetryAddress=192.170.154.204 --set global.remoteZipkinAddress=192.170.154.233 --set sidecarInjectorWebhook.enabled=false --set global.controlPlaneSecurityEnabled=true --set global.mtls.enabled=true --set security.selfSigned=false --set global.hub=docker.io/tiswanso --set global.tag=e2e_pilot_multiclus 
      ```

    1.   **NOTE** the `security.selfSigned=false`--the cacerts secret is created as follows on the remote cluster (same *.pem files as main cluster):
          ```
           kubectl create secret generic cacerts -n istio-system --from-file=samples/certs/ca-cert.pem --from-file=samples/certs/ca-key.pem --from-file=samples/certs/root-cert.pem --from-file=samples/certs/cert-chain.pem
          ```
    

**Version**
*  Istio version = master as of July 2, 2018
*  Kubernetes version on both clusters:

    ```
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.7", GitCommit:"dd5e1a2978fd0b97d9b78e1564398aeea7e7fe92", GitTreeState:"clean", BuildDate:"2018-04-19T00:05:56Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.5", GitCommit:"32ac1c9073b132b8ba18aa830f46b77dcceb0723", GitTreeState:"clean", BuildDate:"2018-06-21T11:34:22Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}

    ```

**Is Istio Auth enabled or not?**
Yes... see above in the steps to reproduce

**Environment**
Baremetal kubernetes.  Multicluster

**Comment by tiswanso** [comment]
*Troubleshooting Summary*

**Comment by tiswanso** [comment]
Yeah... looks like spiffe on remote is bad like you say.  This looks due to using the IP in the pilotDiscoveryAddress.

It's getting the `170` from line 165 here (170 is the 2nd octet of IP addr--gets into var discoveryHostname in this code section): 
https://github.com/istio/istio/blob/master/pilot/cmd/pilot-agent/main.go#L155-L167

Will try to test with a selectorless service for pilot on the remote and use a regular `istio-pilot` hostname in the MeshConfig.

**Comment by tiswanso** [comment]
The selectorless service and endpoint for pilot on the remote and changing the MeshConfig pilotAddress to istio-pilot.istio-system worked.

**Comment by costinm** [comment]
The right fix is to change main.go in agent to not assume the discovery address is istio-pilot.istio-system.
It doesn't make sense to allow address customization if you can't use IP or domain. 

I can make a quick fix, seems easy. This avoids the need for selectorless service.

**Comment by tiswanso** [comment]
Not sure how you'd have enough info to build the value for `verify_subject_alt_name` if you don't have the pilot hostname and namespace in the envoy config.  The value has to match a subjectname in the certificate [1].  It's possible we could remove the check via not adding `verify_subject_alt_name` if the discoveryAddress is an IP.  But not sure we want to try to do that for 1.0.

[1] https://www.envoyproxy.io/docs/envoy/v1.5.0/api-v2/sds.proto#certificatevalidationcontext

**Comment by tiswanso** [comment]
Added doc PR for setup and example:  https://github.com/istio/istio.github.io/pull/1833

**Comment by tiswanso** [comment]
Closing... Fixes and documentation have been merged in release-1.0.
